### ISSUE TITLE
Improve install speed

### ISSUE THREAD

**Issue by brancz** [google-github-actions/setup-gcloud#635]
### TL;DR

Installing `gcloud` (and `gke-gcloud-auth-plugin`) adds about 1 minute to every single one of our CI jobs.

<img width="694" alt="Screenshot 2023-05-04 at 13 40 47" src="https://user-images.githubusercontent.com/4546722/236193918-27574965-bb0a-4f2d-b7e1-11699cbc96fc.png">

Could this be done faster somehow? 1 minute seems rather excessive, or is there a way to skip installing `gcloud` altogether maybe (it appears to be taking the majority of the time)? The only thing we use `gcloud` for is to retrieve the kubeconfig with `gcloud container clusters get-credentials "$GKE_CLUSTER" --zone "$GKE_ZONE"` (we use Workload Identity Federation, so maybe we can just synthesize the kubeconfig?)

**Comment by brancz** [comment]
Apologies for opening this issue, I think I just found the answer myself. I was looking for the combination of:

* https://github.com/google-github-actions/get-gke-credentials
* https://github.com/simenandre/setup-gke-gcloud-auth-plugin

Which only generates the kubeconfig and installs the auth plugin, skipping gcloud all together.
